packages/fx-core/src/plugins/resource/appstudio/plugin.ts

```diff
@@ -345,6 +356,260 @@ export class AppStudioPluginImpl {
+      const hasMsgExt = capabilities?.includes(MessageExtensionItem.id);
+      if (!hasBot && !hasMsgExt) {
+        return err(
+          AppStudioResultFactory.SystemError(
+            AppStudioError.InternalError.name,
+            AppStudioError.InternalError.message
+          )
+        );
+      }
+    }
+    const maybeConfig = this.getConfigForCreatingManifest(ctx, false);
+    if (maybeConfig.isErr()) {
+      return err(maybeConfig.error);
+    }
+
+    const { tabEndpoint, tabDomain, aadId, botDomain, botId, webApplicationInfoResource } =
+      maybeConfig.value;
+
+    const validDomains: string[] = [];
+
+    if (tabDomain) {
+      validDomains.push(tabDomain);
+    }
+
+    if (botDomain) {
+      validDomains.push(botDomain);
+    }
+
+    return ok(
+      this.getDevAppDefinition(
+        JSON.stringify(manifest),
+        aadId,
+        validDomains,
+        webApplicationInfoResource,
+        false,
+        tabEndpoint,
+        manifest.name.short,
+        manifest.version,
+        botId
+      )
+    );
+  }
+
+  /**
+   * The assumptions of this function are:
+   * 1. this.manifest is not undefined(for azure projects) already contains the latest manifest(loaded via reloadManifestAndCheckRequiredFields)
+   * 2. provision of frontend hosting is done and config values has already been loaded into ctx.config
+   * @param ctx
+   * @param maybeSelectedPlugins
+   * @returns
+   */
+  public async createAndConfigTeamsManifest(
+    ctx: PluginContext,
+    maybeSelectedPlugins: Result<LoadedPlugin[], FxError>
+  ): Promise<Result<IAppDefinition, FxError>> {
+    const maybeManifest = await this.reloadManifestAndCheckRequiredFields(ctx.root);
+    if (maybeManifest.isErr()) {
+      return err(maybeManifest.error);
+    }
+    const manifest = maybeManifest.value;
+
+    let appDefinition: IAppDefinition;
+    let updatedManifest: TeamsAppManifest;
+    if (this.isSPFxProject(ctx)) {
+      appDefinition = this.convertToAppDefinition(manifest, false);
+      updatedManifest = manifest;
+    } else {
+      const result = this.createManifestForRemote(ctx, maybeSelectedPlugins, manifest);
+      if (result.isErr()) {
+        return err(result.error);
+      }
+      [appDefinition, updatedManifest] = result.value;
+    }
+
+    const teamsAppId = ctx.config.get(REMOTE_TEAMS_APP_ID) as string;
+    if (!teamsAppId) {
+      ctx.logProvider?.info(`Teams app not created`);
+      const appStudioToken = await ctx?.appStudioToken?.getAccessToken();
+      const result = await this.updateApp(
+        appDefinition,
+        appStudioToken!,
+        "remote",
+        true,
+        undefined,
+        ctx.logProvider,
+        ctx.root
+      );
+      if (result.isErr()) {
+        return result.map((_) => appDefinition);
+      }
+
+      ctx.logProvider?.info(`Teams app created ${result.value}`);
+      appDefinition.appId = result.value;
+      ctx.config.set(REMOTE_TEAMS_APP_ID, result.value);
```

Comment:
@1yefuwang1 , will this cause compatibility issue?

This config is moved from solution to plugin, then the configurations in env.default.json file is changed. So the old project may be effected?

Reply:
That's a good point. Maybe we can find a way to sync appstudioPlugin.REMOTE_TEAMS_APP_ID to solution.REMOTE_TEAMS_APP_ID to maintain backward compatibility.

Reply:
I add REMOTE_TEAMS_APP_ID to solution after the appstudio plugin write to the config.
```ts
        const result = await appStudioPlugin.createAndConfigTeamsManifest(
          pluginCtx,
          maybeSelectedPlugins
        );
        if (result.isErr()) {
          return err(result.error);
        }
        ctx.config.get(GLOBAL_CONFIG)?.set("remoteTeamsAppId", result.value.appId);
```